test/api.jl

```diff
@@ -0,0 +1,39 @@
+module APITests
+
+using Pkg, Test
+
+@testset "Pkg.activate" begin
```

Comment:
This testset exactly mirrors the one for activate in repl.jl, but with using the API. A bit unfortunate, would be nice to tests equivalent things like this in the same place